### PR TITLE
Refactor border.error.color to also accept as error.border.color

### PR DESCRIPTION
**Pull request by g4rry420**
#### What does this PR do?
border.error.color to also accept as error.border.color

#### Where should the reviewer start?
components => FormField.js

#### What testing has been done on this PR?
yarn test-update

#### How should this be manually tested?
enter `error: { border: { color: "colorName" } } ` in custom theme

#### Any background context you want to provide?
N/A

#### What are the relevant issues?
#4004 

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
Yes, I think for also mentioning `error.border.color` in `base.d.ts`

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Comment by britt6612** [comment]
@g4rry420 this looks good! Can you also add this to the docs so users can know that `formfield` will accept both..

**Comment by g4rry420** [comment]
Done @britt6612

**Comment by britt6612** [comment]
@g4rry420 Nice looks good to me! Ill let Shimi or Eric review now!

**Review by jcfilben — COMMENTED** [review]
Looks good! I am wondering if in the documentation we should mark `border.error.color` as deprecated and recommend the use of `error.border.color` instead?

**Review by ericsoderberghp — APPROVED** [review]

File: src/js/components/FormField/doc.js [hunk 1]
```diff
@@ -159,6 +159,11 @@ export const themeDoc = {
     type: 'string | boolean | number',
     defaultValue: undefined,
   },
+  'formField.error.border.color': {
+    description: 'The border color of the error.',
```
Comment (ShimiSun) [inline src/js/components/FormField/doc.js:163]: Let's add a deprecation message to the deprecated style.

**Comment by ericsoderberghp** [comment]
Thanks for your contribution!

**Pull request merged**